### DISCUSSION TITLE
[documentation] how to use backbutton dispatcher

### DISCUSSION THREAD

**Discussion by aoatmon** [csells/go_router#101] [Q&A] [answered]
Hello,
maybe it's just me, 
but I cannot find any reference on if / how GoRouter provides a back button dispatcher

**Comment by csells** [comment]
It does not. Is that something you need it to do? What problem are you trying to solve?

**Comment by FabianVarela** [comment, marked as answer]
@aoatmon I made this implementation for BackButtonDispatcher and it works for me.
 
```dart
home: Router<dynamic>(
     routerDelegate: _router.routerDelegate,
     backButtonDispatcher: RootBackButtonDispatcher(),
),
```

**Reply by ebelevics** [reply]
thank you! this works like a charm for nested routers

**Comment by csells** [comment]
interesting. @FabianVarela @aoatmon what are you attempting to accomplish with the `BackButtonDispatcher` that `GoRouter` doesn't do by default?

**Comment by mrseth01** [comment]
@csells I'm confused. You wrote that you need to provide a back button dispatcher. And then, you said Go Router does this by default. 

How do I navigate back? I don't see a "goBack" or "pop" method on GoRouter? What is the best way to go back programmatically?

**Comment by csells** [comment]
Did I say that you needed to provide a `BackButtonDispatcher`? Can you point out where I said that so I can fix it?

`Navigator.maybePop(context)` will go back programmatically.

**Comment by aoatmon** [comment]
@csells @FabianVarela

[RootBackButtonDispatcher](https://api.flutter.dev/flutter/widgets/RootBackButtonDispatcher-class.html) does what I need it to do, thank you both
